pkg/jetstream/publisher.go

```diff
@@ -0,0 +1,85 @@
+package jetstream
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ThreeDotsLabs/watermill"
+	wmnats "github.com/ThreeDotsLabs/watermill-nats/v2/pkg/nats"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var _ message.Publisher = &Publisher{}
+
+type Publisher struct {
+	nc     *nats.Conn
+	js     jetstream.JetStream
+	m      wmnats.Marshaler
```

Comment:
This is only dependency on nats pkg maybe worth duplicating.  Odd marshaler/unmarshal we situation here.